Update Parsing and Quoting articles (#8053)
MicrosoftDocs/PowerShell-Docs: reference/7.2/Microsoft.PowerShell.Core/About/about_Parsing.md

    `$HOME.Length` and string literal `-more`.

- Quotation marks (`'` and `"`) begin strings

- Braces (`{}`) demarcate new script blocks

- Commas (`,`) introduce lists passed as arrays, except when the command to be
  called is a native application, in which case they are interpreted as part of
  the expandable string. Initial, consecutive or trailing commas are not
  supported.

- Parentheses (`()`) demarcate new expressions
- Subexpression operator (`$()`) demarcates embedded expressions
- Initial at sign (`@`) begins expression syntaxes such as splatting (`@args`),
  arrays (`@(1,2,3)`), and hash table literals (`@{a=1;b=2}`).
- `()`, `$()`, and `@()` at the start of a token create a new parsing context
  that can contain expressions or nested commands.
  - When followed by additional characters, the first additional character is
    considered the start of a new, separate argument.
  - When preceded by an unquoted literal `$()` works like an expandable string,
    `()` starts a new argument that is an expression, and `@()` is taken as
    literal `@` with `()` starting a new argument that is an expression.

> - Everything else is treated as an expandable string, except that
>   metacharacters still need escaping.

Reviewer: ```suggestion
- Everything else is treated as an expandable string, except
  metacharacters that still need escaping.
```

icacls X:\VMS /grant Dom\HVAdmin:`(CI`)`(OI`)F
```

### The stop-parsing token

> Beginning in PowerShell 3.0, you can use the stop-parsing token (`--%`). The
> stop-parsing token directs PowerShell to refrain from interpreting input as
> PowerShell commands or expressions.

Reviewer: ```suggestion
Beginning in PowerShell 3.0, you can use the stop-parsing token (`--%`) to
stop PowerShell from interpreting input asPowerShell commands or expressions.
```

> [!NOTE]

> > The stop-parsing token only intended for use on Windows platforms.

Reviewer: ```suggestion
> The stop-parsing token is only intended for use on Windows platforms.
```
